pkg/client/interfaces.go

```diff
@@ -44,10 +45,36 @@ type Patch interface {
 	Type() types.PatchType
 	// Data is the raw data representing the patch.
 	Data(obj runtime.Object) ([]byte, error)
+	// DataFromUnstructured is the raw data representing the patch from unstructured
+	DataFromUnstructured(obj Unstructured) ([]byte, error)
 }
 
 // TODO(directxman12): is there a sane way to deal with get/delete options?
 
+// Unstructured is an interface to type unstructured.Unstructured in apimachinery
+type Unstructured interface {
```

Comment:
why aren't we just using `runtime.Unstructured` here?

Reply:
coz we only use `SetUnstructuredContent(map[string]interface{})` and `Object` interface from `runtime.Unstructured`. since it will save us two lines of code, i will include it.